bootstrap.py

```diff
@@ -19,10 +19,71 @@ def eprint(msg):
     #  the bootstrap.
     # Keep in sync with `slm.toml`
     DEPENDENCIES = {
-        "stanza-toml": {"git": "StanzaOrg/stanza-toml", "version": "0.4.0"},
-        "maybe-utils": {"git": "StanzaOrg/maybe-utils", "version": "0.1.4"},
-        "semver": {"git" : "StanzaOrg/semver", "version" : "0.1.6"},
-        "term-colors": { "git": "StanzaOrg/term-colors", "version" : "0.1.1"},
+        "curl_static":              { "git": "StanzaOrg/slm_curl",              "version": "0.0.1" },
```

Comment:
It is really annoying that we have to duplicate this data. I think this is getting to the point where it is actively harmful.

The only reason we are doing this is because Python doesn't natively support `toml` before version 3.11. I think the options are:

1.  Support only Python 3.11 or greater
2.  Move to `JSON` instead of `TOML`
3.  Add a python toml parser implementation as a file in this local directory and then use that to parse the `slm.toml` file.
4.  Add a `requirements.txt` file for the bootstrap and use a venv to install a `toml` dependency for python 3 less than py3.11.

It seems like with the introduction of `conan` there is `requirements.txt` file already. You are now using the `requests` dependency (among possibly others) in the `bootstrap.py` file. Perhaps Option 4 above is the right way to go - we can use some toml dependency to parse the `slm.toml` file and remove this static, duplicate data.

Reply:
I'm inclined to support only Python 3.12+.  Since we're using TOML, it seems reasonable to require having TOML built-in in order to build.  If an advanced user really wanted to use 3.11, they could install tomllib themselves.

Reply:
So I vote for removing this version check and hack from this file.  :smile: